other_data_collections/2015ApJ...812...60B/Add_Biteau.py

```diff
@@ -0,0 +1,190 @@
+# This script reads in Biteau's catalog stored in 
+# BiteauWilliams2015_AllData_ASDC_v2016_12_20.ecsv,
+# it creates the corresponding ecsv-sed files and stores
+# them in /input/data
+
+from astropy.table import Table
+from astropy.io import ascii
+from astropy.constants import h,e
+from pathlib import Path
+import astropy.units as u
+import numpy as nump
+import os
+import glob
+
+class Biteau:
+    fBiteauFile = './BiteauWilliams2015_AllData_ASDC_v2016_12_20.ecsv'
+    fBiteauSourceIDFile = './Biteau_Sources.txt'
+    fBiteauExperimentFile = './Biteau_Experiments.txt'
+    fInputDataPath = './../../input/data/'
+
+    def __init__(self):
+        self.fBiteauCatalog = self.Load_Catalog(self.fBiteauFile)
+        self.fBiteauSources = self.Load_SourceIDs(self.fBiteauSourceIDFile)
+        self.fBiteauExperiments = self.Load_Experiments(self.fBiteauExperimentFile)
+
+# Load_Catalog reads Biteau's catalog
+    def Load_Catalog(self, filename):
+        print('Reading {}'.format(filename))
+        biteau_catalog = Table.read(filename, format = 'ascii.ecsv', delimiter='|')
+        return biteau_catalog
+
+# Load_SourceIDs read Biteau_Sources.txt in which all sources which are mentioned in
+# Biteau's catalog and the corresponding source_id are stored
+    def Load_SourceIDs(self, filename):
+        biteau_sources = dict()
+        print('Reading {}'.format(filename))
+        data = ascii.read(filename)
+        for i in range(0, len(data)):
+            biteau_sources[data[i][0]] = data[i][1]
+        return biteau_sources
+
+# Load_Experiments read Biteau_Experiments.txt in which all sources which are mentioned in
+# Biteau's catalog and the corresponding telescope from /input/schemas are stored
+    def Load_Experiments(self, filename):
+        biteau_experiments = dict()
+        print('Reading {}'.format(filename))
+        data = ascii.read(filename)
+        for i in range(0, len(data)):
+            biteau_experiments[data[i][0]] = data[i][1]
+        return biteau_experiments
+
+# Rename_Experiment renames all source names in Biteau's catalog into the corresponding name
+# in /input/schemas
+    def Rename_Experiment(self, experiment):
+        return self.fBiteauExperiments[experiment]
+
+# Calculate_Energy transforms the 'freq'-column in Biteau's catalog into an energy-column 
+    def Calculate_Energy(self, frequency):
+        energy = (h*frequency*1/u.second)/e.si * 1E-9* u.teraelectronvolt/u.joule*u.coulomb
+        return energy
+
+# Calculate_dnde transforms the 'e2dnde'-column in Biteau's catalog into an dnde-column
+    def Calculate_dnde(self, energy, e2dnde):
```

Comment:
The functions you have to compute `energy` and `dnde` can be improved:
* Make them standalone functions above your class (methods that take `self` as argument but then don't use it aren't nice)
* I don't really understand why the code you have to compute these is such a long line, what I have in https://nbviewer.jupyter.org/gist/cdeil/2942d2f0241d7ff8d25db8008e0fda03 is shorter / more readable, no?
In any case, there's still "magic constants" like 1e9 and 1e-16 in the formula, that shouldn't be necessary if you use quantities. And the line for `dnde` is too long, I have to scroll back and forth on Github to review it. So bottom line: please try to make it more readable, using the code in the notebook I linked to as example.